### DISCUSSION TITLE
2.0.0-Pre 问题反馈

### DISCUSSION THREAD

**Discussion by am-abudu** [Qexo/Qexo#174] [Announcements]
1. 重新设计文章编辑器
2. 登录/友链申请验证码支持
3. 重构图床/服务商组件架构
4. 优化设置界面用户体验
5. 添加SMTP邮件推送和PushDeer推送
6. 支持将推送内容转为Markdown格式
7. 修复大量错误; 更新依赖
8. 较为完整的自部署支持

WARN: 本次更新可能会导致图床配置丢失, 请注意备份

TIPS: Preview 更新可能存在较多 BUG, 如遇问题可积极参与讨论

<hr /><em>This discussion was created from the release <a href='https://github.com/Qexo/Qexo/releases/tag/2.0.0-Pre'>2.0.0-Pre</a>.</em>

**Comment by imjason** [comment]
文章没有修改也有弹出对话框

**Reply by Fgaoxing** [reply]
这个就是这样的

**Reply by Fgaoxing** [reply]
没问题

**Reply by am-abudu** [reply]
目前没有对这个进行判断，所以是正常现象

**Reply by am-abudu** [reply]
已完成修改

**Comment by Hulenkius** [comment]
刚刚我通过Qexo设置页面中的“一键更新”更新了2.0.0-Pre，文章编辑页面没有显示编辑框，出现了如下报错：

ScannerError('while scanning for the next token', None, "found character '\t' that cannot start any token", <yaml.error.Mark object at 0x7fae7c23a130>)

我尝试手动更新，出现了同样的错误。

**Reply by am-abudu** [reply]
应该是部分文章吧，我认为是个Bug，将尝试修复。是否可分享问题文章的源文件供测试？

**Reply by Hulenkius** [reply]
问题页面是我点击 “新文章” 后出现的页面。
我猜想，可能跟我的 Hexo 的 scaffolds/post.md 有关，它的内容如下：

```
---
title: 
top: false
math: false
hidden: false
theme: flip	#不要更改這一行！
widgets: 'null'	#不要更改這一行！
date: {{ date }}
author: 
cover: 
thumbnail: 
category: 
password: 
---
```

**Reply by am-abudu** [reply]
@Hulenkius 看报错信息是`\r`制表符的问题, 尝试删除`theme: flip	#不要更改這一行！`中flip后面的部分？

**Reply by Hulenkius** [reply]
好，我测试一下

**Reply by Hulenkius** [reply]
经测试，删除上述部分后已经可以正常编辑文章，问题解决。

**功能性建议：**

我的 Hexo 博客是多人编辑的，这使得文件头中的 author 等参数的可定制性十分重要；而且我们还商量好不使用 tag 参数。
Qexo 2.0.0-Pre 版在文章编辑页面右边添加了标题、发布于、更新于、标签、分类五个基本参数的输入框，将其他参数的自定义放到了下方的“自定义”输入框，对我们来说造成了一定不便。

能否实现自定义“文章编辑页”右边的“文章参数”内容呢？

**Reply by am-abudu** [reply]
@Hulenkius 如果没有问题的话，下个版本尝试支持

**Reply by am-abudu** [reply]
> 经测试，删除上述部分后已经可以正常编辑文章，问题解决。
> 
> **功能性建议：**
> 
> 我的 Hexo 博客是多人编辑的，这使得文件头中的 author 等参数的可定制性十分重要；而且我们还商量好不使用 tag 参数。 Qexo 2.0.0-Pre 版在文章编辑页面右边添加了标题、发布于、更新于、标签、分类五个基本参数的输入框，将其他参数的自定义放到了下方的“自定义”输入框，对我们来说造成了一定不便。
> 
> 能否实现自定义“文章编辑页”右边的“文章参数”内容呢？

自定义侧边栏 a9b66c61fbadf38e4706faba5411765acd3a1825

**Reply by Hulenkius** [reply]
打开任一文章编辑页面后不显示编辑框，点击侧边栏“文章参数”的编辑按钮没有反应。

按F12观测到的错误信息：
开启页面后：
`Uncaught SyntaxError: Unexpected token ';' (at new.html:900:26)`

点击编辑按钮后：
`Uncaught ReferenceError: editSide is not defined`
    `at <anonymous>:1:1`

![image](https://user-images.githubusercontent.com/32562298/180610665-9b4e024a-6e49-415d-b77c-aada3f06127c.png)

**Reply by am-abudu** [reply]
@Hulenkius 在更新Dev分支后，因为版本号并没有改变，所以程序无法自动更新配置，所以你需要手动访问一下`/update/`以执行配置更新

**Reply by Hulenkius** [reply]
~~已解决~~？

**Reply by Hulenkius** [reply]
旧参数可以删除，无法新建、修改参数。

**Reply by am-abudu** [reply]
@Hulenkius 我这里是可以的，是否可看一下报错

**Reply by Hulenkius** [reply]
https://user-images.githubusercontent.com/32562298/180611716-cb4b62aa-bb51-4fd4-a3bb-5353659d081d.mp4

**Reply by am-abudu** [reply]
我了解了，明天在测试修复吧，今天有点肝不动了

**Reply by am-abudu** [reply]
@Hulenkius 不修好还是睡不好，刚刚给修了，这次测试过没问题了

**Reply by Hulenkius** [reply]
**新发现问题：**
文章参数中需要填 true 或 false 的，通过侧边栏编辑的结果会自动加上单引号，变成字符串，导致 'false' 被自动识别为 true，参数无法发挥作用。

**功能性建议：**
侧边栏加入 checkbox，来规定参数布尔值。
可以实现为：勾选即为 true，不勾选即不填（自动加入两个单引号不影响布尔值判断）

**Reply by am-abudu** [reply]
这样有些麻烦，我直接对true flase进行特殊判定可否

**Reply by Hulenkius** [reply]
也行

**Reply by Hulenkius** [reply]
再加一个 null

**Reply by am-abudu** [reply]
27b0f909450ca236a88741c9c3db078daf27f44c

**Comment by qvsijia** [comment]
虽然我啥也不懂，但是这个项目很好

**Comment by Fgaoxing** [comment]


**Reply by am-abudu** [reply]
我这里测试可以正常使用，你需要在文章模板里面添加{ abbrlink }
